src/Compiler/SyntaxTree/SyntaxTree.fs

```diff
@@ -908,6 +908,8 @@ type SynPat =
 
     | Or of lhsPat: SynPat * rhsPat: SynPat * range: range * trivia: SynPatOrTrivia
 
+    | Cons of lhsPat: SynPat * rhsPat: SynPat * range: range * trivia: SynPatConsTrivia
```

Comment:
Let's call it `ListCons` or something like that? `Cons` is a bit too general and this name could imply something like a union case constructor too.